chore(docs): simplify using-gatsby-image doc (#11998)
gatsbyjs/gatsby: docs/docs/using-gatsby-image.md

`gatsby-image` is designed to work seamlessly with Gatsby’s native image processing capabilities powered by GraphQL and Sharp. To produce perfect images, you only need to:

1. Install `gatsby-image` and other, necessary dependencies like `gatsby-plugin-sharp` and `gatsby-transformer-sharp`

```shell
  npm i gatsby-image gatsby-transformer-sharp gatsby-plugin-sharp
```

2. Add the newly installed plugins and transformer plugins to your `gatsby-config.js`

```js:title=gatsby-config.js
module.exports = {
  plugins: [`gatsby-plugin-sharp`, `gatsby-transformer-sharp`],
}
```

> 3. Configure `gatsby-source-filesystem` to load images from a folder. In order to use GraphQL to query the image files, the files need to be in a location that is known to Gatsby. This requires an to `gatsby-config.js` to configure the plugin. Feel free to replace the `path` option with wherever your images are located relative to your project.

Reviewer: >This requires an to `gatsby-config.js`...

I think this needs a word like `entry` or `edit` here, i.e. "This requires an edit to `gatsby-config.js...`"

```js:title=gatsby-config.js
module.exports = {
  plugins: [
    `gatsby-transformer-sharp`,
    `gatsby-plugin-sharp`
    // highlight-start
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        path: `${__dirname}/src/data/`,
      }
    }
    // highlight-end
  ],
}
```

4. Write a GraphQL query using one of the included [GraphQL “fragments”](/packages/gatsby-image/#fragments) which specify the fields needed by `gatsby-image` to create a responsive, optimized image. This example will use `GatsbyImageSharpFluid`. An example of a GraphQL query is below where the path listed is the path relative to the location specified in the `gatsby-source-filesystem` options.

```graphql
file(relativePath: { eq: "images/default.jpg" }) {
      childImageSharp {
        # Specify the image processing specifications right in the query.
        fluid {
          ...GatsbyImageSharpFluid
        }
      }
}
```

5. Import `Img` to display the fragment in JSX. There are additional features available with the `Img` tag as well.

```jsx
import Img from "gatsby-image"

export default ({ data }) => (
  <div>
    <h1>Hello gatsby-image</h1>

>     <Img fluid={data.file.childImageSharp.fluid} />

Reviewer: Please add `alt` text here to set a good example for accessibility. You can get pretty creative with it! 😃

`<Img fluid={data.file.childImageSharp.fluid} alt="JAM stack jammies" />`

  </div>
)
```

> This GraphQL query creates multiple sizes of the image. The image that is appropriate for the current screen resolution (e.g. desktop, mobile, and everything in between). The `gatsby-image` component automatically enables a blur-up effect as well as lazy loading images that are not currently on screen.

Reviewer: The second sentence here isn't quite a full sentence. Here's an easy fix: "Gatsby will load the image that is appropriate for the current screen resolution (e.g. desktop, mobile, and everything in between)."

`gatsby-image` is designed to work seamlessly with Gatsby’s native image processing capabilities powered by GraphQL and Sharp. To produce perfect images, you only need to:

> 1. Install `gatsby-image` and its peer dependencies. The `gatsby-config.json` needs to include the following:

Reviewer 2: ```suggestion
1. Install `gatsby-image` and other, necessary dependencies like `gatsby-plugin-sharp` and `gatsby-transformer-sharp`
    ```shell
    npm i gatsby-image gatsby-transformer-sharp gatsby-plugin-sharp
    ```
1. Add the newly installed plugins and transformer plugins to your `gatsby-config.js`
    ```js:title=gatsby-config.js
    module.exports = {
      plugins: [
        // highlight-start
        `gatsby-plugin-sharp`,
        `gatsby-transformer-sharp`
      ]
    }
    ```
```

```js:title=src/gatsby-config.js
module.exports = {
  plugins: [
    // highlight-start

>     `gatsby-image`,

Reviewer 2: Not necessary to add gatsby-image here!

```suggestion
```

Reviewer 2: Actually - let's just strike this whole section! It's encapsulated in the suggestion above ☝️

    `gatsby-transformer-sharp`,
    `gatsby-plugin-sharp`,
    // highlight-end
  ],
}
```

> 2. Configure the peer dependencies to load images from a folder. In order to use GraphQL to query the image files, the files need to be in a location that is known to Gatsby. This requires another update to `gatsby-config.json`. Any path can replace `src/data` in this example, but whatever the path is the image file being queried needs to be placed within the scope of this path.

Reviewer 2: ```suggestion
2. Configure `gatsby-source-filesystem` to load images from a folder. In order to use GraphQL to query the image files, the files need to be in a location that is known to Gatsby. This requires an to `gatsby-config.js` to configure the plugin. Feel free to replace the `path` option with wherever your images are located relative to your project.
```

```js:title=gatsby-config.js
module.exports = {
  plugins: [
    `gatsby-image`,
    `gatsby-transformer-sharp`,
    `gatsby-plugin-sharp`
    // highlight-start
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        path: `${__dirname}/src/data/`,
      }
    }
    // highlight-end
  ],
}
```

> 3. Write a GraphQL query using one of the included GraphQL “fragments” which specify the fields needed by `gatsby-image`. There are numerous fragments that can be used and the choice is based on the type of responsive image desired. This example will use `GatsbyImageSharpFluid`. An example of a GraphQL query is below where the path listed is the path relative to the location specified in the `gatsby-source-filesystem` configuration.

Reviewer 2: ```suggestion
3. Write a GraphQL query using one of the included [GraphQL “fragments”](/packages/gatsby-image/#fragments) which specify the fields needed by `gatsby-image` to create a responsive, optimized image. This example will use `GatsbyImageSharpFluid`. An example of a GraphQL query is below where the path listed is the path relative to the location specified in the `gatsby-source-filesystem` options.
```

```graphql
file(relativePath: { eq: "images/default.jpg" }) {
      childImageSharp {
        # Specify the image processing specifications right in the query.
        fluid {
          ...GatsbyImageSharpFluid
        }
      }
}
```

4. Import `Img` to display the fragment in JSX. There are additional features available with the `Img` tag as well.

```jsx
import Img from "gatsby-image"

export default ({ data }) => (
  <div>
    <h1>Hello gatsby-image</h1>
    <Img fluid={data.file.childImageSharp.fluid} />
  </div>
)
```

> The GraphQL query creates multiple thumbnails with optimized JPEG and PNG compression. The `gatsby-image` component automatically sets up the “blur-up” effect as well as lazy loading of images further down the screen.

Reviewer 2: ```suggestion
This GraphQL query creates multiple sizes of the image. The image that is appropriate for the current screen resolution (e.g. desktop, mobile, and everything in between). The `gatsby-image` component automatically enables a blur-up effect as well as lazy loading images that are not currently on screen.
```
